### PR TITLE
Make Link component dynamic based on loggedIn

### PR DESCRIPTION
**Pull request by clarkmcadoo**
Added the LoggedInProvider to list of items being exported from the theme and added that provider to the demo page.

Updated Link component to see if urls are product URLs, if they are, and the user is not logged it, it will send them to the signup page.  If they are logged in, it will send them to the product.

### CODE REVIEW

File: packages/gatsby-theme-newrelic/src/components/Link.js [hunk 1]
```diff
@@ -46,6 +47,16 @@ const Link = forwardRef(
       }
     `);
 
+    // Supplied from LoggedInProvider
+    const { loggedIn } = useLoggedIn();
+
+    if (to.includes('one.newrelic.com')) {
+      // Using session storage, so the value will come as a string.
+      if (loggedIn === 'false') {
```
Comment (LizBaker) [inline packages/gatsby-theme-newrelic/src/components/Link.js:55]: i think the useLoggedIn hook always returns a boolean and not a string

Comment (clarkmcadoo) [reply packages/gatsby-theme-newrelic/src/components/Link.js:55]: i was having a lot of trouble getting it to work just via a Boolean and idk why. i'll try again

**Review by LizBaker — APPROVED** [review]

File: packages/gatsby-theme-newrelic/src/hooks/useLoggedIn.js [hunk 2]
```diff
@@ -55,8 +55,6 @@ export const useLoggedIn = () => {
   return { loading, loggedIn };
 };
 
-export default useLoggedIn;
```
Comment (sunnyzanchi) [inline packages/gatsby-theme-newrelic/src/hooks/useLoggedIn.js:58]: i missed this the first time but `gatsby-theme-newrelic/index.js` reexports the default export from this file as `useLoggedIn`. i pushed `71e5a45` to fix this

**Review by sunnyzanchi — APPROVED** [review]
✨

**Pull request merged**